opm/material/fluidsystems/BlackOilFluidSystem.hpp

```diff
@@ -476,7 +518,40 @@ class BlackOil : public BaseFluidSystem<Scalar, BlackOil<Scalar> >
      */
     static bool enableVaporizedOil()
     { return enableVaporizedOil_; }
+    /*!
+     * \brief Returns the density of a fluid phase at surface pressure [kg/m^3]
+     *
+     * \copydoc Doxygen::phaseIdxParam
+     */
+    static Scalar rateLimmitDissolvedGas()
+    { return rateLimmitDissolvedGas_;}
+
+    template<class LHS>
+    static LHS rateLimmitedUpdate(LHS So,Scalar So0,LHS RsSat,Scalar RsSat0,double dt){
+        auto drs_max=rateLimmitDissolvedGas()*dt;
+        //auto So_tmp = So0;// we do this explicitely to avoid high derivatives.
+
+        if((RsSat-RsSat0) >= drs_max){
+            RsSat = RsSat0 + drs_max;
```

Comment:
the functionality of this method is probably better moved to `Ewoms::BlackOilIntensiveQuantities::update()`, `Ewoms::BlackPrimaryVariables::adaptPrimaryVariables()` and `Ewoms::BlackNewtonMethod::update()`.